# MODIS Fire Dataset (India, 2024)

This dataset comes from **NASA FIRMS (Fire Information for Resource Management System)**, which uses the **MODIS satellite instruments (Aqua and Terra)** to detect active fire hotspots across the globe.

###  Dataset Overview
- **Source:** NASA FIRMS (MODIS Collection)
- **Region:** India
- **Year:** 2024
- **Format:** CSV
- **Captured by:** MODIS satellites (Aqua & Terra)

###  What Each Row Represents
Each row in the dataset corresponds to a **fire hotspot detected by satellite**.
It contains information such as:
- **Latitude & Longitude** - location of the fire pixel
- **Brightness** - intensity of the fire
- **Confidence** - reliability of detection (low, nominal, high)
- **Acquisition Date & Time** - when the fire was observed
- **Satellite** - which satellite captured the data (Aqua or Terra)
- **FRP (Fire Radiative Power)** - energy released by the fire
- **Day/Night flag** - whether detection was during day or night

###  Why This Dataset Matters
This dataset is directly relevant to **CERT‑Sat**, since fires are a **threat signal**.
By analyzing this data, we can:
- Detect unusual fire patterns (anomalies).
- Identify regions with high fire risk.
- Train models to automatically flag potential threats.


In [ ]:
# Let's understand important points from above.

### 1. Confidence in Fire Detection

- **Confidence** indicates how reliable the satellite detection of a fire hotspot is.
- It is categorized into:
  - **Low** - weak or uncertain detection
  - **Nominal** - medium certainty, reasonably reliable
  - **High** - strong signal, almost certainly a real fire
- This helps to decide whether to trust a detection or filter out possible false alarms.
- In CERT‑Sat, confidence acts like a **trust score** for threat signals.


### 2. Aqua and Terra Satellites

- **Terra** (launched in 1999) and **Aqua** (launched in 2002) are NASA satellites that orbit Earth in a sun‑synchronous path.
- Both carry the **MODIS instrument (Moderate Resolution Imaging Spectroradiometer)**, which scans Earth in multiple wavelengths.

- **Terra** crosses the equator in the morning (10:30 a.m.), while **Aqua** crosses in the afternoon (1:30 p.m.).

- Together, they provide **global coverage every 1–2 days**, capturing data on fires, vegetation, oceans, clouds, and climate.

- In our dataset:
  - Rows marked **Terra** = fire detected during the morning pass.
  - Rows marked **Aqua** = fire detected during the afternoon pass.


### 3. Fire Radiative Power (FRP)

- **FRP** measures the energy released by a fire, based on thermal radiation detected by MODIS.
- **Unit:** Megawatts (MW).
- **Interpretation:**
  - Higher FRP - stronger, more intense fire.
  - Lower FRP - smaller or less severe fire.
- **Importance:**
  - Indicates fire severity and biomass burning.
  - Helps estimate emissions (CO₂, aerosols).
  - Acts as a key **threat signal** in CERT‑Sat analysis.


In [ ]:
# Moving further.

## Step 2: Inspecting Dataset Structure and Quality

Before analyzing fire patterns, we need to understand the dataset’s structure and check its quality.
- **Shape:** How many rows and columns are present.
- **Columns:** Names of all variables (like brightness, confidence, FRP).
- **Data types:** Whether values are numbers, text, or dates.
- **Missing values:** Identify if any data is incomplete.
- **Duplicates:** Check if the same fire record appears more than once.
- **Summary statistics:** Get basic measures (mean, min, max) to understand ranges.


In [6]:
import pandas as pd

# Loaded  dataset
df = pd.read_csv("modis_2024_India.csv")

df.info()


<class 'pandas.DataFrame'>
RangeIndex: 74029 entries, 0 to 74028
Data columns (total 15 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   latitude    74029 non-null  float64
 1   longitude   74029 non-null  float64
 2   brightness  74029 non-null  float64
 3   scan        74029 non-null  float64
 4   track       74029 non-null  float64
 5   acq_date    74029 non-null  str    
 6   acq_time    74029 non-null  int64  
 7   satellite   74029 non-null  str    
 8   instrument  74029 non-null  str    
 9   confidence  74029 non-null  int64  
 10  version     74029 non-null  float64
 11  bright_t31  74029 non-null  float64
 12  frp         74029 non-null  float64
 13  daynight    74029 non-null  str    
 14  type        74029 non-null  int64  
dtypes: float64(8), int64(3), str(4)
memory usage: 8.5 MB


###  Dataset Structure

- **Total records (rows):** 74,029
- **Total variables (columns):** 15

### Insights
- Dataset is **complete**: all 74,029 entries have non‑null values.
- Mix of **numeric (float/int)** and **categorical (string)** variables.
- Variables cover **location, timing, intensity, reliability, and classification** — giving a full picture of fire events.
- Memory usage: ~8.5 MB, which is manageable for analysis on a laptop.

In [18]:
df.columns

Index(['latitude', 'longitude', 'brightness', 'scan', 'track', 'acq_date',
       'acq_time', 'satellite', 'instrument', 'confidence', 'version',
       'bright_t31', 'frp', 'daynight', 'type'],
      dtype='str')

###  Column Details

- **latitude, longitude (float)** - location of fire hotspot
- **brightness (float)** - intensity of fire pixel
- **scan, track (float)** - pixel size in scan/track direction
- **acq_date (string)** - date of detection
- **acq_time (int)** - time of detection (UTC, HHMM format)
- **satellite (string)** - Aqua or Terra
- **instrument (string)** - MODIS instrument
- **confidence (int)** - reliability of detection (low/nominal/high, stored as numbers)
- **version (float)** - dataset processing version
- **bright_t31 (float)** - brightness temperature at 31 microns
- **frp (float)** - Fire Radiative Power (MW)
- **daynight (string)** - detection during day or night
- **type (int)** - fire type classification

In [17]:
df.head()

,latitude,longitude,brightness,scan,track,acq_date,acq_time,satellite,instrument,confidence,version,bright_t31,frp,daynight,type
0,23.7690,86.3923,303.8,3.6,1.8,2024-01-01,346,Terra,MODIS,13,61.03,291.9,27.5,D,2
1,34.0996,74.9958,305.7,1.0,1.0,2024-01-01,522,Terra,MODIS,55,61.03,278.0,10.1,D,0
2,31.0805,78.2029,300.8,1.5,1.2,2024-01-01,522,Terra,MODIS,18,61.03,284.3,11.5,D,0
3,34.0908,74.9942,300.5,1.0,1.0,2024-01-01,522,Terra,MODIS,28,61.03,280.4,6.9,D,0
4,23.6291,74.3663,304.3,1.2,1.1,2024-01-01,524,Terra,MODIS,43,61.03,292.2,5.1,D,0


###  Why This Matters

Looking at the first few rows helps to confirm following:
- The dataset is loading correctly.
- The values make sense (dates, times, satellite names).
- We can already see variation in fire intensity and confidence levels.

In [21]:
df.describe()

,latitude,longitude,brightness,scan,track,acq_time,confidence,version,bright_t31,frp,type
count,74029.000000,74029.000000,74029.000000,74029.000000,74029.000000,74029.000000,74029.000000,7.402900e+04,74029.000000,74029.000000,74029.000000
mean,23.782620,81.170065,322.040333,1.389465,1.141921,871.591633,63.700199,6.103000e+01,301.930673,24.841774,0.140648
std,5.061686,6.300820,14.037963,0.599460,0.193873,381.909639,17.975680,1.421095e-14,8.239977,73.853068,0.513286
min,8.159800,68.532300,300.000000,1.000000,1.000000,303.000000,0.000000,6.103000e+01,267.400000,0.000000,0.000000
25%,20.799900,76.287300,312.200000,1.000000,1.000000,717.000000,54.000000,6.103000e+01,296.600000,8.000000,0.000000
50%,23.928400,79.516700,320.100000,1.100000,1.100000,827.000000,65.000000,6.103000e+01,301.000000,12.400000,0.000000
75%,27.227500,84.012900,329.500000,1.500000,1.200000,859.000000,75.000000,6.103000e+01,307.500000,22.200000,0.000000
max,34.649400,97.078300,509.600000,4.800000,2.000000,2212.000000,100.000000,6.103000e+01,400.100000,7528.500000,3.000000


### Summary Statistics

The dataset provides basic measures for numeric columns:

- **Latitude & Longitude**
  - Range: Latitude ~8.16° to 34.65°, Longitude ~68.53° to 97.07°
  - Insight: Covers the geographic extent of India, confirming dataset region.

- **Brightness**
  - Mean: ~322
  - Min: 300, Max: 509.6
  - Insight: Most fires are moderate intensity, but some extreme hotspots exist.

- **Scan & Track (pixel size)**
  - Typical values: 1.0–1.5
  - Max scan: 4.8, Max track: 2.0
  - Insight: Pixel resolution varies slightly, affecting fire area estimates.

- **Acquisition Time (UTC)**
  - Range: 303 to 2212 (HHMM format)
  - Insight: Fires detected across different times of day.

- **Confidence**
  - Mean: ~63.7
  - Range: 0 to 100
  - Insight: Most detections are medium to high confidence, but some low‑certainty records exist.

- **Version**
  - Constant: 61.03
  - Insight: All records processed with the same dataset version.

- **Brightness T31**
  - Mean: ~302, Range: 267.4 to 400.1
  - Insight: Indicates thermal variation in fire pixels.

- **FRP (Fire Radiative Power)**
  - Mean: ~24.8 MW
  - Range: 0 to 7528.5 MW
  - Insight: Most fires are small (FRP < 25), but rare extreme events release massive energy.

- **Type**
  - Mean: ~0.14, Range: 0 to 3
  - Insight: Majority of records are type 0 (default classification), with few higher codes.


In [14]:
 print('Missing Values:\n')
 df.isnull().sum()


Missing Values:



latitude      0
longitude     0
brightness    0
scan          0
track         0
acq_date      0
acq_time      0
satellite     0
instrument    0
confidence    0
version       0
bright_t31    0
frp           0
daynight      0
type          0
dtype: int64

### Missing Values

- Result: **No missing values** in any of the 15 columns.
- Every record has complete information for location, brightness, confidence, FRP, and other variables.

### Insights
- The dataset is **clean and reliable** — we don’t need to perform imputation (filling gaps) or drop incomplete rows.
- This makes analysis easier and ensures that all fire detections are fully described.
- Having complete data increases confidence in the results of our EDA and later modeling.


In [16]:
print("Duplicate rows:\n", df.duplicated().sum())


Duplicate rows:
 0


###  Duplicate Records

- Result: **0 duplicate rows** found in the dataset.
- Every fire detection record is unique — no repeated entries.

###  Insights
- This confirms the dataset is **clean and reliable**.
- We don’t need to remove any duplicates, so analysis will not be biased by repeated fire detections.
- Each of the 74,029 records represents a distinct fire hotspot observed by MODIS.
